examples/sql/client-api/advanced_query_options/main.go

```diff
@@ -31,42 +32,58 @@ func main() {
 		panic(fmt.Errorf("creating the client"))
 	}
 	defer client.Shutdown(ctx)
-	opts := hazelcast.SQLOptions{}
-	opts.SetSchema("partitioned")
-	_, err = client.ExecSQLWithOptions(ctx, `
+	stmt := sql.NewStatement(`
 			CREATE OR REPLACE MAPPING mymap
 			TYPE IMAP
 			OPTIONS (
 				'keyFormat' = 'bigint',
 				'valueFormat' = 'varchar'
 			)
-	`, opts)
+	`)
+	stmt.SetSchema("partitioned")
+	sqlService := client.SQL()
+	_, err = sqlService.ExecuteStatement(ctx, stmt)
 	if err != nil {
 		panic(fmt.Errorf("creating the mapping: %w", err))
 	}
-	stmt, err := client.PrepareSQL(ctx, `SINK INTO mymap VALUES(?, ?)`)
-	if err != nil {
-		panic(fmt.Errorf("preparing statement: %w", err))
-	}
-	defer stmt.Close()
 	for i := 0; i < 100; i++ {
-		if _, err = stmt.ExecContext(ctx, i, fmt.Sprintf("sample string-%d", i)); err != nil {
+		if _, err = sqlService.Execute(ctx, `SINK INTO mymap VALUES(?, ?)`,
+			i, fmt.Sprintf("sample string-%d", i)); err != nil {
 			panic(fmt.Errorf("inserting values: %w", err))
 		}
 	}
-	opts.SetQueryTimeout(5 * time.Second)
-	opts.SetCursorBufferSize(10)
-	rows, err := client.QuerySQLWithOptions(ctx, `SELECT __key, this from mymap order by __key`, opts)
+	stmt = sql.NewStatement(`SELECT __key, this from mymap order by __key`)
+	stmt.SetQueryTimeout(5 * time.Second)
+	stmt.SetCursorBufferSize(10)
+	result, err := sqlService.ExecuteStatement(ctx, stmt)
 	if err != nil {
 		panic(fmt.Errorf("querying: %w", err))
 	}
-	defer rows.Close()
-	var k int64
-	var v string
-	for rows.Next() {
-		if err := rows.Scan(&k, &v); err != nil {
-			panic(fmt.Errorf("scanning: %w", err))
+	defer result.Close()
+	it, err := result.Iterator()
+	if err != nil {
+		panic(fmt.Errorf("acquiring iterator: %w", err))
+	}
+	var (
+		k   int64
+		v   string
+		tmp interface{}
```

Comment:
`tmp` is not a very good name,

Reply:
How about `iValue`?

Reply:
Why don't we call it `v` or even better `value`? I don't think `v = tmp.(string)` line is necessary.

Reply:
Removed type assertions all together. They are provided in employee example anyway
https://github.com/hazelcast/hazelcast-go-client/pull/755/commits/26ded879d4df2716aa36029e4a5a7412f0602524